activerecord/test/cases/attribute_methods_test.rb

```diff
@@ -1246,6 +1246,92 @@ class ChildWithDeprecatedBehaviorResolved < ClassWithDeprecatedAliasAttributeBeh
     assert_equal("overridden_subject_was", obj.subject_was)
   end
 
+  ClassWithGeneratedAttributeMethodTarget = Class.new(ActiveRecord::Base) do
+    self.table_name = "topics"
+    alias_attribute :saved_title, :title_in_database
+  end
+
+  test "#alias_attribute with an _in_database method issues a deprecation warning" do
+    message = <<~MESSAGE.gsub("\n", " ")
+    AttributeMethodsTest::ClassWithGeneratedAttributeMethodTarget model aliases `title_in_database`, but title_in_database is not an attribute.
+    Starting in Rails 7.2 `, alias_attribute with non-attribute targets will raise. Use `alias_method :saved_title`, :title_in_database or define the method manually.
+    MESSAGE
+
+    obj = assert_deprecated(message, ActiveRecord.deprecator) do
+      ClassWithGeneratedAttributeMethodTarget.new
+    end
+    obj.title = "A river runs through it"
+    assert_nil obj.saved_title
+    obj.save
+    assert_equal "A river runs through it", obj.saved_title
+  end
+
+  ClassWithEnumMethodTarget = Class.new(ActiveRecord::Base) do
+    self.table_name = "books"
+
+    attribute :status, :string
+    enum status: {
+      pending: "0",
+      completed: "1",
+    }
+    alias_attribute :is_pending?, :pending?
+  end
+
+  test "#alias_attribute with enum method issues a deprecation warning" do
+    message = <<~MESSAGE.gsub("\n", " ")
+    AttributeMethodsTest::ClassWithEnumMethodTarget model aliases `pending?`, but pending? is not an attribute.
+    Starting in Rails 7.2 `, alias_attribute with non-attribute targets will raise.
+    Use `alias_method :is_pending?`, :pending? or define the method manually.
+    MESSAGE
+
+    obj = assert_deprecated(message, ActiveRecord.deprecator) do
+      ClassWithEnumMethodTarget.new
+    end
+    obj.status = "pending"
+    assert obj.pending?
```

Comment:
A nitpick but `assert_predicate` generally raises a better error message like `"expected <obj> to be `pending?`"` while `assert` prints a very dry "expected false to be truthy"

```suggestion
    assert_predicate obj, :pending?
```

Reply:
That's much nicer, thanks! Updated as part of [this change](https://github.com/rails/rails/compare/02eaf46709ceabb9eabcee6ed995439ef632f7c0..22f8548bea2127222fddaf84b2488c4b93661742)